tests/phpunit/integration/Modules/Analytics_4/SettingsTest.php

```diff
@@ -33,16 +34,85 @@ public function test_get_default() {
 				// TODO: These can be uncommented when Analytics and Analytics 4 modules are officially separated.
 				// 'accountID'       => '',
 				// 'adsConversionID' => '',
-				'propertyID'      => '',
-				'webDataStreamID' => '',
-				'measurementID'   => '',
-				'useSnippet'      => true,
-				'ownerID'         => 0,
+				'propertyID'           => '',
+				'webDataStreamID'      => '',
+				'measurementID'        => '',
+				'useSnippet'           => true,
+				'ownerID'              => 0,
+				'googleTagID'          => '',
+				'googleTagAccountID'   => '',
+				'googleTagContainerID' => '',
 			),
 			get_option( Settings::OPTION )
 		);
 	}
 
+	public function test_owner_id_is_set() {
+		$testcase = $this->get_testcase();
+		$settings = $this->get_setting_with_owned_keys();
+		$settings->register();
+
+		$user_id = $testcase->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+		// Ensure admin user has Permissions::MANAGE_OPTIONS cap regardless of authentication.
+		add_filter(
+			'map_meta_cap',
+			function( $caps, $cap ) {
+				if ( Permissions::MANAGE_OPTIONS === $cap ) {
+					return array( 'manage_options' );
+				}
+				return $caps;
+			},
+			99,
+			2
+		);
+
+		$options_key = $testcase->get_option_name();
+		$fields      = $settings->get_owned_keys();
+		foreach ( $fields as $field ) {
+			delete_option( $options_key );
+
+			$options = $settings->get();
+			$testcase->assertEmpty( $options['ownerID'] );
+
+			if ( 'googleTagID' === $field ) {
+				$options[ $field ] = 'G-XXXXXX';
+			} else {
+				$options[ $field ] = 'test-value';
+			}
+			$settings->set( $options );
+
+			$options = get_option( $options_key );
+			$testcase->assertEquals( $user_id, $options['ownerID'] );
+		}
+	}
+
+	public function test_validate_google_tag_id() {
+		$testcase = $this->get_testcase();
+		$settings = $this->get_setting_with_owned_keys();
+		$settings->register();
+
+		$options_key = $testcase->get_option_name();
+
+		delete_option( $options_key );
+
+		$options                = $settings->get();
+		$options['googleTagID'] = 'XXX';
+		$settings->set( $options );
+		$options = get_option( $options_key );
+		$testcase->assertEquals( '', $options['googleTagID'] );
+
+		$google_tag_ids_valid = array( 'G-XXXX', 'GT-XXXX', 'AW-XXXX' );
+
+		foreach ( $google_tag_ids_valid as $tag_id ) {
```

Comment:
Please use a `dataProvider` instead. This way we can test all the elements in the set independently. Done within a single test, it will only go as far as the first failing assertion. A provider offers a bit more context in the output as well which is useful for debugging a failure 👍

Reply:
@aaemnnosttv makes sense. Do you have an example of such a `dataProvier` from another test?